spec/features/female_missing_info_feature_spec.rb

```diff
@@ -119,4 +127,50 @@
       wait_for { page.current_path == missing_information_prison_prisoners_path(prison.code) }
     end
   end
+
+  context 'when multiple tabs are opened', :js do
+    let(:complexity) { nil }
+
+    it 'can complete forms in parallel' do
+      visit missing_information_prison_prisoners_path prison.code
+
+      within "#edit_#{prisoner_id}" do
+        click_link 'Add missing details'
+      end
+
+      find('label[for=complexity-form-complexity-level-medium-field]').click
+
+      click_button 'Update'
+
+      fill_in_missing_case_information
+      click_button 'Update'
+
+
+      execute_in_new_tab do
+        visit missing_information_prison_prisoners_path prison.code
+
+        within "#edit_#{second_prisoner_id}" do
+          click_link 'Add missing details'
+        end
+
+        find('label[for=complexity-form-complexity-level-high-field]').click
+
+        click_button 'Update'
+
+        fill_in_missing_case_information
+
+        click_button 'Update'
+      end
```

Comment:
Unless I'm misunderstanding what's going on here – aren't these two things happening one after the other? It doesn't look like they're happening in parallel any more.

Really we need both tabs to be on the "Add missing details" page **at the same time** – and then "Update" one before the other. Ultimately: replicating the behaviour of a user who opens multiple tabs and then updates them one by one.

I think without that, we're not actually proving that the bug is fixed..?

Happy to be corrected if I've got that wrong!

Reply:
Yes you're right @ollietreend the second journey should happen *after* the first one (which is what it used to do)

Reply:
Oh I see. I have put it in the wrong order! Thank you @ollietreend